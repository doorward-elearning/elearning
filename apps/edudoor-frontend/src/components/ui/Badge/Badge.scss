.ed-badge {
  background: var(--accent);
  color: var(--text-alternate);
  padding: 0 var(--padding);
  border-radius: 10px;
}
