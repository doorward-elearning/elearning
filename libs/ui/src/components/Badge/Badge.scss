.ed-badge {
  background: var(--accent);
  color: var(--text-alternate);
  padding: var(--padding-sm) var(--padding);
  border-radius: 5px;
}
