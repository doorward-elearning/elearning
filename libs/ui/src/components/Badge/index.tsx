import React from 'react';
import './Badge.scss';
import classNames from 'classnames';

const Badge: React.FunctionComponent<BadgeProps> = props => {
  return (
    <span
      className={classNames({
        'ed-badge': true,
      })}
    >
      {props.children}
    </span>
  );
};

export interface BadgeProps {}

export default Badge;
