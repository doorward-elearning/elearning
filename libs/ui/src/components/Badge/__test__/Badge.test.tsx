import * as React from 'react';
import { shallow } from 'enzyme';
import Badge from '../index';
import '@testing-library/jest-dom';
import '../Badge.scss';
import classNames from 'classnames';

describe('Badge component test', () => {
  const iconValue = <img alt="Test icon" />;
  // Snapshot Tests
  it('should render a default component with nothing', () => {
    const wrapper = shallow(<Badge />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a badge component with theme ', () => {
    const wrapper = shallow(<Badge theme="success" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a badge component with classname', () => {
    const wrapper = shallow(<Badge className='ed-badge' />);
    expect(wrapper).toMatchSnapshot();
  });
  //prop Tests
  it('should have the class ed-badge ', () => {
    const wrapper = shallow(<Badge />);
    expect(wrapper.prop('className')).toContain('ed-badge');
  });

  it('should have the class ed-badge along with render class ed-badge', () => {
    const wrapper = shallow(<Badge className="ed-badge" />);
    expect(wrapper.prop('className')).toContain('ed-badge ed-badge');
  });
  
  it('should have the class ed-badge along with theme', () => {
    const wrapper = shallow(<Badge theme='success' />);
    expect(wrapper.prop('className')).toContain('ed-badge success');
  });
});

