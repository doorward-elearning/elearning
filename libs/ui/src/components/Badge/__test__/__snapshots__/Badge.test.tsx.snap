// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component test should render a badge component with classname 1`] = `ShallowWrapper {}`;

exports[`Badge component test should render a badge component with theme  1`] = `ShallowWrapper {}`;

exports[`Badge component test should render a default component with nothing 1`] = `ShallowWrapper {}`;

exports[`Badge with only icon should render a badge component with classname 1`] = `ShallowWrapper {}`;

exports[`Badge with only icon should render a badge component with theme  1`] = `ShallowWrapper {}`;

exports[`Badge with only icon should render a default component with nothing 1`] = `ShallowWrapper {}`;

exports[`should render a badge component with theme  1`] = `ShallowWrapper {}`;

exports[`should render a default component with nothing 1`] = `ShallowWrapper {}`;
